Switchboard 1.4.0 release

jrhender opened the pull request:
Changes are:
- JSON field type -> **needs ssi-hub  v1.12 release to production** https://github.com/energywebfoundation/ssi-hub/pull/576
- SIWE login

nichonien committed:
chore: update asset manager address (#822)

This PR sets AssetManager contract address directly (rather than from iam-client-lib) as a temporary fix to unblock Asset functionality.

michalziobro committed:
fix: upgrade bootstrap from 4.3.1 to 4.6.2 (#809)

Snyk has created this PR to upgrade bootstrap from 4.3.1 to 4.6.2.

See this package in npm:
https://www.npmjs.com/package/bootstrap

See this project in Snyk:
https://app.snyk.io/org/iam-team-bca/project/a7d263b1-45ed-4f30-8f8f-d1b1940fe65f?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

michalziobro committed:
fix: upgrade @sentry/angular from 6.17.8 to 6.19.7 (#808)

Snyk has created this PR to upgrade @sentry/angular from 6.17.8 to 6.19.7.

See this package in npm:
https://www.npmjs.com/package/@sentry/angular

See this project in Snyk:
https://app.snyk.io/org/iam-team-bca/project/a7d263b1-45ed-4f30-8f8f-d1b1940fe65f?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

michalziobro committed:
fix: upgrade @fortawesome/fontawesome-free from 5.5.0 to 5.15.4 (#806)

Snyk has created this PR to upgrade @fortawesome/fontawesome-free from 5.5.0 to 5.15.4.

See this package in npm:
https://www.npmjs.com/package/@fortawesome/fontawesome-free

See this project in Snyk:
https://app.snyk.io/org/iam-team-bca/project/a7d263b1-45ed-4f30-8f8f-d1b1940fe65f?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

nichonien committed:
chore: update iam-client-lib to v7.0.0

nichonien committed:
fix: revert asset manager address hotfix

nichonien committed:
Merge pull request #823 from energywebfoundation/revert_asset_manager_hotfix

fix: revert asset manager hotfix

michalziobro committed:
fix: upgrade ngx-toastr from 14.2.1 to 14.3.0 (#829)

Snyk has created this PR to upgrade ngx-toastr from 14.2.1 to 14.3.0.

See this package in npm:
https://www.npmjs.com/package/ngx-toastr

See this project in Snyk:
https://app.snyk.io/org/iam-team-bca/project/a7d263b1-45ed-4f30-8f8f-d1b1940fe65f?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

dawidgil committed:
chore: rename button with 'claim' into 'enrolment' (#824)

dawidgil committed:
feat(sentry): add performance tracing

* feat(sentry): add performance tracing

* fix: setting severity level for iam logger

snyk-bot committed:
fix: upgrade rxjs from 7.5.6 to 7.8.0

Snyk has created this PR to upgrade rxjs from 7.5.6 to 7.8.0.

See this package in npm:
https://www.npmjs.com/package/rxjs

See this project in Snyk:
https://app.snyk.io/org/iam-team-bca/project/a7d263b1-45ed-4f30-8f8f-d1b1940fe65f?utm_source=github&utm_medium=referral&page=upgrade-pr

dawidgil committed:
fix: package-lock file conflicts

dawidgil committed:
feat: add json schema to role definition fields

* feat: allow to create JSON field for requestor and issuer

* feat: handle json in textarea

* feat: use json editor for adding schema

* refactor: remove nesting fields properties

* feat: handle setting json schema while enrolling

* fix: prettier

* fix: tests

* fix: eslint errors

* fix: breaking changes in ICL

* fix: no need to stringify json field value

* fix: validating data depending on schema

* fix: package-lock json file

* feat(json-editor): better styling

* feat: create pipe for displaying json schema value

* feat: stringify json data

* fix: prettier

* fix(ci): allocating memory for unit tests

michalziobro committed:
fix: upgrade simple-line-icons from 2.4.1 to 2.5.5

* fix: upgrade simple-line-icons from 2.4.1 to 2.5.5

Snyk has created this PR to upgrade simple-line-icons from 2.4.1 to 2.5.5.

See this package in npm:
https://www.npmjs.com/package/simple-line-icons

See this project in Snyk:
https://app.snyk.io/org/iam-team-bca/project/a7d263b1-45ed-4f30-8f8f-d1b1940fe65f?utm_source=github&utm_medium=referral&page=upgrade-pr

* fix: package lock file

Co-authored-by: snyk-bot <snyk-bot@snyk.io>
Co-authored-by: dawidgil <dawid.gil@j-labs.pl>

whitneypurdum committed:
feat: set Issuer Request issuer fields with merge button (from requestor fields) (#833)

* feat: set form fields with input from props

* fix: remove logs

* feat: add test

dawidgil committed:
fix: reduce number of calls while creating enrolment claim

* fix: reduce number of calls while creating enrolment claim

* test: user effects

* refactor: clean up not needed code

* fix: prettier

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/energywebfoundation/switchboard-dapp?pullRequest=839) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/energywebfoundation/switchboard-dapp?pullRequest=839) before we can accept your contribution.<br/>**7** out of **8** committers have signed the CLA.<br/><br/>:white_check_mark: nichonien<br/>:white_check_mark: dawidgil<br/>:white_check_mark: michalziobro<br/>:white_check_mark: whitneypurdum<br/>:white_check_mark: jrhender<br/>:white_check_mark: Timtech4u<br/>:white_check_mark: JGiter<br/>:x: snyk-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/energywebfoundation/switchboard-dapp?pullRequest=839) it.</sub>

jrhender committed:
feat(deps): icl to 7.1.0-alpha.3 (#840)

michalziobro committed:
fix: package.json & package-lock.json to reduce vulnerabilities 

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-DECODEURICOMPONENT-3149970

Co-authored-by: snyk-bot <snyk-bot@snyk.io>

dawidgil committed:
fix: improve performance of getting enrolments

* feat: improve performance while getting enrolments

* feat: remove enrolment from my list when canceling request

* refactor: remove console logs

* refactor: create base class for enrolment effects

* test: remove code deprecations

* test(owned-enrolments): selectors and reducer

* test: selectors for enrolments

* refactor: use id instead of enrolment in actions and effects

* fix: prettier

* refactor: add returning types to claim facades

* test: update enrolment effects

dawidgil committed:
feat: improve enrolments nats updates

* fix: improve performance for enrolment update after getting nats notifications

* fix: updating issuer list after canceling claim

* fix: tests in message subscription

* fix: prettier

* fix: restore removed action

* refactor: create enrolments facade for handling enrolments state

* test: enrolments facade

* fix: prettier

Timtech4u committed:
chore: setup switchboard deployment for gp4btc-staging & prod (#837)

* chore: setup gp4btc-staging deploy

* chore: remove unused branch

* feat: deployment scripts for gp4btc staging and prod

* build(cf-deploy): gp4btc staging & prod build

* build(cf-deploy): add gp4btc-prod as valid branch

* style(cf-deploy): apply prettier

---------

Co-authored-by: dawidgil <dawid.gil@j-labs.pl>
Co-authored-by: John Henderson <john.henderson@energyweb.org>

dawidgil committed:
fix: error with closeConnection method call when iam service isn't initialised

dawidgil committed:
fix(issuer-requests): throwing errors when opening view

JGiter committed:
feat: update iam-client-lib@8.0.0-alpha.1 (#847)

Initial SIWE support

JGiter committed:
build: update iam-client-lib@8.0.0-alpha.2

JGiter committed:
Merge pull request #850 from energywebfoundation/update_iam-client-lib@8.0.0-alpha.2

update iam-client-lib@8.0.0-alpha.2

dawidgil committed:
fix: error with getting allowed roles by issuer

dawidgil committed:
fix: close connection error, as signerService might not be initialised

whitneypurdum committed:
fix: revert change to make fields static (#851)

jrhender committed:
feat(iam.service): set domain to window origin (#853)

jrhender committed:
build(gp4btc prod): fix ICS and NATS URL (#855)

dawidgil committed:
fix: disable verbose while starting app locally

dawidgil committed:
chore(build): enable source maps for all builds

dawidgil committed:
chore(deps): update sentry to v7.43

whitneypurdum committed:
fix: fix development file replacement (#861)

dawidgil committed:
chore(deps): update angular and related packages to v14

* chore(deps): update angular and related packages to v14

* fix: prettier

* fix: revert changes in angular json file

* refactor: remove Date creation from expiration date

* revert: remove Date creation from expiration date, adjust types

* fix(request-claim): type of roleType

* fix: prettier

dawidgil committed:
chore: add husky git hooks

dawidgil committed:
chore(build): send source maps to sentry

* fix: dev configuration

* feat: send source maps to sentry

* refactor: adjust configuration

whitneypurdum committed:
feat: add button styling to merge button (#868)

dawidgil committed:
fix: commitlint config

dawidgil committed:
chore(deps): update to angular to v15

* chore(deps): update angular and related packages to v15

* fix: bootstrap imports

* fix: material components

* fix: package lock file

* fix: imports

* fix: styling of legacy components

* fix: prettier

* fix: tests for user selectors

* fix: organization selectors tests

* fix: requested selectors

* fix: owned selectors tests

* fix: auth selectors tests

* fix: owned assets selectors tests

* fix: multiple component and service tests

* fix: tests for sb toastr

* fix: message subscription service tests

* fix: prettier

* refactor: use spying for message subscription

* fix: creating eslint report as output file

dawidgil committed:
fix: build scripts

dawidgil committed:
fix(ci): node version

dawidgil committed:
fix: better displaying of json schema in details

dawidgil committed:
fix: json schema validation in enrolment form

dawidgil committed:
fix: better displaying of json schema in details

* refactor: rename input to better display what it does

* fix: not aligned requestor fields in details

* fix: prettier

* fix: add small padding between field value and merge button

dawidgil committed:
chore(deps): update all possible dependencies

dawidgil committed:
fix(ci): build script for gp4btc stg env (#877)

dawidgil committed:
fix: check schema validation after merging

dawidgil committed:
refactor: remove sending issuer dids while creating role

* refactor: remove sending issuer dids while creating role

* refactor: remove casting to any

dawidgil committed:
fix: problem with navigating to org creation enrolment (#874)
